### PR TITLE
Alethe: Translate REORDERING rule

### PR DESCRIPTION
**Pull request by Lachnitt**
Implementation of the translation of REORDERING rules into the Alethe calculus.

Should be merged after PR #7532.

Co-authored-by: Haniel Barbosa hanielbbarbosa@gmail.com

### CODE REVIEW

File: src/proof/alethe/alethe_post_processor.cpp [hunk 1]
```diff
@@ -668,6 +668,34 @@ bool AletheProofPostprocessCallback::update(Node res,
...
+    case PfRule::REORDERING:
+    {
+      Node trueNode = nm->mkConst(true);
+      std::vector<Node> new_children = children;
+      if (children[0].getKind() == kind::OR
+          && (args[0] != trueNode || children[0] != args[1]))
+      {
+        std::shared_ptr<ProofNode> childPf = cdp->getProofFor(children[0]);
+        if (childPf->getRule() == PfRule::ASSUME
+            || childPf->getRule() == PfRule::EQ_RESOLVE)
```
Comment (HanielB) [inline src/proof/alethe/alethe_post_processor.cpp:684]: you'd have to remove this to generalize, right?

Comment (Lachnitt) [reply src/proof/alethe/alethe_post_processor.cpp:684]: Yes, I handed this in before the meeting yesterday and wanted to ask before changing this.

Comment (Lachnitt) [reply src/proof/alethe/alethe_post_processor.cpp]: Done

**Comment by Lachnitt** [comment]
As discussed in the proofs meeting I took out the part where an or step was added.

**Review by HanielB — APPROVED** [review]

**Pull request merged**